### PR TITLE
[JSDoc] Add mising documentation for botbuilder-testing

### PR DESCRIPTION
**Pull request by sw-joelmut**
Addresses # 2602

## Description
This PR adds all the missing JSDoc documentation for [botbuilder-testing](https://github.com/microsoft/botbuilder-js/tree/main/libraries/botbuilder-testing) based on the errors thrown. Some documentation is ported and adapted from [botbuilder-dotnet](https://github.com/microsoft/botbuilder-dotnet).

## Specific Changes

- Ported documentation from botbuilder-dotnet
- Added `@private` for private methods.
- Added JSDoc comments in **2 files**.

## Testing
In the following image there is a sneak peek about the ESLint Report with no JSDoc errors.
image

### CODE REVIEW

**Review by santgr11 — APPROVED** [review]
Since all the comments we are adding in the rest of PRs are using `Optional,` instead of `(Optional)` I would like to change these to be the same as the rest, but I'm not sure if this is the proper approach since this comment is one that already existed that is being copied to the overloads.

The rest lgtm

**Comment by sw-joelmut** [comment]
@santgr11 , fixed, thanks

**Review by iscaltritti-sw — APPROVED** [review]

File: libraries/botbuilder-testing/src/dialogTestClient.ts [hunk 1]
```diff
@@ -29,25 +29,44 @@ export class DialogTestClient {
...
+     * Creates a `DialogTestClient` to test a dialog without having to create a full-fledged adapter.
+     * ```javascript
+     * let client = new DialogTestClient(MY_DIALOG, MY_OPTIONS);
+     * let reply = await client.sendActivity('first message');
+     * assert.strictEqual(reply.text, 'first reply', 'reply failed');
+     * ```
+     * @param testAdapter A list of middlewares to be added to the test adapter.
+     * @param targetDialog The dialog to be tested. This will be the root dialog for the test client.
+     * @param initialDialogOptions Optional. Additional argument(s) to pass to the dialog being started.
+     * @param middlewares Optional. The test adapter to use. If this parameter is not provided, the test client will use a default TestAdapter.
```
Comment (ceciliaavila) [inline libraries/botbuilder-testing/src/dialogTestClient.ts:58]: The descriptions of testAdapter and middlewares are mixed.

File: libraries/botbuilder-testing/src/dialogTestClient.ts [hunk 2]
```diff
@@ -29,25 +29,44 @@ export class DialogTestClient {
...
+     * @param conversationState Optional. A ConversationState instance to use in the test client.
+     */
     public constructor(testAdapter: TestAdapter, targetDialog: Dialog, initialDialogOptions?: any, middlewares?: Middleware[], conversationState?: ConversationState)
+    /**
+     * Creates a `DialogTestClient` to test a dialog without having to create a full-fledged adapter.
+     * @param channelOrAdapter The channelId to be used for the test or a list of middlewares to be added to the test adapter.
+     * @param targetDialog The dialog to be tested. This will be the root dialog for the test client.
+     * @param initialDialogOptions Optional. Additional argument(s) to pass to the dialog being started.
+     * @param middlewares Optional. The test adapter to use. If this parameter is not provided, the test client will use a default TestAdapter.
```
Comment (ceciliaavila) [inline libraries/botbuilder-testing/src/dialogTestClient.ts:67]: Same in here.

**Review by ceciliaavila — CHANGES_REQUESTED** [review]

**Comment by sw-joelmut** [comment]
@ceciliaavila, feedback applied, thanks

**Review by ceciliaavila — APPROVED** [review]

**Review by iscaltritti-sw — APPROVED** [review]

**Review by denscollo — APPROVED** [review]

**Review by ceciliaavila — APPROVED** [review]

**Comment by denscollo** [comment]
Promoted

**Pull request closed without merging**